server/fishtest/rundb.py

```diff
@@ -1091,7 +1093,22 @@ def count_games(d):
             task["active"] = False
             return {"task_alive": False, "error": error}
 
-        # The update seems fine. Update run["tasks"][task_id] (=task).
+        # The update seems fine.
+
+        # Increment run results before overwriting task["stats"]
```

Comment:
This should be equivalent (btw iterate with the built-in dict iterator `for s in stats:` it's faster than iterate on the view of the keys `for s in stats.key():`).
```python
for key, value in stats.items():
    if key == "pentanomial":
        run["results"][key] = [
            x + y - z
            for x, y, z in zip(run["results"][key], value, task["stats"].get(key, [0] * 5))
        ]
    else:
        run["results"][key] += value - task["stats"].get(key, 0)
```

Reply:
Yes, I like this.

I should have remembered that `.items()` was an option, and it works nicely here, e.g. just makes that last line a bit simpler.

I was looking at zip() more when I coded `check_results()`. When I was coding this pentanomial branch, it seemed that zip would be more complicated so I just did my loop thing.